src/main/java/jenkins/plugins/git/GitSSHPrivateKeyBinding.java

```diff
@@ -0,0 +1,217 @@
+        Git gitInstance = Git.with(listener, env).using(gitToolExe);
+        return gitInstance.getClient();
+    }
+
+    @Override
+    protected Class<SSHUserPrivateKey> type() {
+        return SSHUserPrivateKey.class;
+    }
+
+    private boolean isGitVersionAtLeast(GitClient git, int major, int minor, int rev, int bugfix) {
+        return ((CliGitAPIImpl) git).isCliGitVerAtLeast(major, minor, rev, bugfix);
+    }
+
+    private String getSSHCmd(SSHUserPrivateKey credentials, FilePath tempDir,String sshExePath) throws IOException, InterruptedException {
+        if (unixNodeType) {
+            return "ssh -i "
+                    + "\""
+                    + getPrivateKeyFile(credentials, tempDir).getRemote()
+                    + "\" "
+                    + "-o StrictHostKeyChecking=no $@";
+        } else {
+            return  "\"" + sshExePath + "\""
+                    + " -i "
+                    + "\""
+                    + getPrivateKeyFile(credentials, tempDir).getRemote()
+                    + "\""
+                    + " -o StrictHostKeyChecking=no";
+        }
+    }
+
+    protected final class SSHScriptFile extends AbstractOnDiskBinding<SSHUserPrivateKey> {
+
+        private final String sshExePath;
+        private final boolean unixNodeType;
+
+        protected SSHScriptFile(SSHUserPrivateKey credentials, String sshExePath, boolean unixNodeType) {
+            super(SSHKeyUtils.getPrivateKey(credentials) + ":" + SSHKeyUtils.getPassphrase(credentials), credentials.getId());
+            this.sshExePath = sshExePath;
+            this.unixNodeType = unixNodeType;
+        }
+
+        @Override
+        protected FilePath write(SSHUserPrivateKey credentials, FilePath workspace) throws IOException, InterruptedException {
+            FilePath tempFile;
+            if (unixNodeType) {
+                tempFile = workspace.createTempFile("gitSSHScript", ".sh");
+                tempFile.write(
+                        "ssh -i "
+                                + getPrivateKeyFile(credentials, workspace).getRemote()
+                                + " -o StrictHostKeyChecking=no $@", null);
+                tempFile.chmod(0500);
+            } else {
+                tempFile = workspace.createTempFile("gitSSHScript", ".bat");
+                tempFile.write("@echo off\r\n"
+                        + "\""
+                        + this.sshExePath
+                        + "\""
+                        + " -i "
+                        + "\""
+                        + getPrivateKeyFile(credentials, workspace).getRemote()
+                        + "\""
+                        + " -o StrictHostKeyChecking=no", null);
+            }
+            return tempFile;
+        }
+
+        @Override
+        protected Class<SSHUserPrivateKey> type() {
+            return SSHUserPrivateKey.class;
+        }
+    }
+
+    @Symbol("GitSSHPrivateKey")
```

Comment:
Need to start the symbol with a lower case letter for consistency
```suggestion
    @Symbol("gitSSHPrivateKey")
```